src/Framework/Config/GacelaFileConfig/Factory/GacelaConfigFromBootstrapFactory.php

```diff
@@ -25,13 +25,16 @@ public function __construct(array $globalServices)
 
     public function createGacelaFileConfig(): GacelaConfigFile
     {
-        /**
-         * @var array{
-         *     config?: callable,
-         *     mapping-interfaces?: callable,
-         *     suffix-types?: callable,
-         * } $configFromGlobalServices
-         */
+        $configBuilder = $this->createConfigBuilder();
+        $mappingInterfacesBuilder = $this->createMappingInterfacesBuilder();
+        $suffixTypesBuilder = $this->createSuffixTypesBuilder();
+
+        return GacelaConfigFile::usingBuilders($configBuilder, $mappingInterfacesBuilder, $suffixTypesBuilder);
+    }
+
+    private function createConfigBuilder(): ConfigBuilder
```

Comment:
What about? (same with others :) )

```php
    private function createConfigBuilder(): ConfigBuilder
    {
        /** @var callable|null $configFromGlobalServices */
        $configFromGlobalServices = $this->globalServices['config'] ?? null;

        $configBuilder = new ConfigBuilder();
        if ($configFromGlobalServices !== null) {
            return $configFromGlobalServices($configBuilder);
        }

        return $configBuilder;
    }
```

Reply:
Interesting, but it wouldn't work because of `return $configFromGlobalServices($configBuilder)`. Thanks for the idea, anyway 💡